_posts/2017-01-21-Wonders-of-Static-Linking.md

```diff
@@ -0,0 +1,147 @@
+---
+layout: post
+title: "Wonders Of Static Linking"
+date: 2017-01-21 20:37:00
+author: lordPoseidon
+---
+
+Hi, this is Nimit Bhardwaj, a student of Computer Science, we all in the journey of computer science got the chance to learn the C Programming language, yes we all, not all although, learning C as out first programming language taught us many things, many things which are related to the working of the basics of the programming.
+
+We are taught how to compile a program, yes very easy process, most of us think its just pressing the **Compile** button or pressing the **Compile and Execute** button for those who use Dev-C/C++ or just use `gcc` who use linux. Its a good defination for a layman, but for a programmer, yes for us, its not enough. For the coders(some basic coders), like us, there are 4 steps of the compilation these are
+
+* Preprocessing
+* Compilation
+* Assembling
+* Linking
+
+The preprocessing is easy for C/C++ languages, it just replace the pre-program directive, the lines starting with `#`, like `#include <stdio.h>` with the file which has the **prototypes** of the functions which will be used or replace the Macros with their expansions like `#define POSEIDON 7`, this line will replace the occurence of POSEIDON with 7, pretty basic huh.
+
+Then is compilation, which checks the possiblities of the syntax and sementic errors if possible and then will initilize the assembling phase of the process.
```

Comment:
> Then is compilation

Something like `The next phase is compilation` sounds better to me. Same thing for the next few paragraphs also.

> sementic

typo, should be semantic

> initilize

initialize